packages/react-router/modules/matchPath.js

```diff
@@ -34,25 +34,31 @@ const matchPath = (pathname, options = {}) => {
     strict = false,
     sensitive = false
   } = options;
-  const { re, keys } = compilePath(path, { end: exact, strict, sensitive });
-  const match = re.exec(pathname);
 
-  if (!match) return null;
+  const paths = [].concat(path);
 
-  const [url, ...values] = match;
-  const isExact = pathname === url;
+  return paths.reduce((matched, path) => {
```

Comment:
Is this the only way to isolate which of the paths in the array matches? It isn't actually taking advantage of `path-to-regexp`'s array support, which I thought was the justification for supporting an array as `path`.

Reply:
I'm not sure if it's the only way, but it struck me as the least disruptive way and the one least likely to introduce new, unforeseen bugs.

The only potential problem I see with this implementation is with performance (e.g., iteration + regex could spell trouble), and I'm happy to look into alternative approaches if this one isn't acceptable. However, my instinct is that the performance should be just fine for any list of paths of reasonable length, and this change should have ~0 impact on consumers who don't pass an array of paths.

Since this does provide a nicer API and behavior (i.e., no remounting) when consumers want to render the same component for multiple paths, I think it's worth merging and dealing with performance issues if/when they come up. That said, if you disagree, I'm happy to look for another approach.

Reply:
Let's go ahead and merge this, and we can look at using `path-to-regexp`' built-in support for arrays later.